jsonnet/components/metrics-server.libsonnet

```diff
@@ -0,0 +1,377 @@
+      selector: {
+        'app.kubernetes.io/name': 'metrics-server',
+      } + cfg.commonLabels,
+    },
+  },
+  deployment: {
+    apiVersion: 'apps/v1',
+    kind: 'Deployment',
+    metadata: {
+      labels: {
+        'app.kubernetes.io/name': 'metrics-server',
+        'app.kubernetes.io/component': 'metrics-server',
+      } + cfg.commonLabels,
+      name: 'metrics-server',
+      namespace: cfg.namespace,
+    },
+    spec: {
+      minReadySeconds: 60,
+      replicas: 2,
+      selector: {
+        matchLabels: {
+          'app.kubernetes.io/name': 'metrics-server',
+          'app.kubernetes.io/component': 'metrics-server',
+        } + cfg.commonLabels,
+      },
+      strategy: {
+        rollingUpdate: {
+          maxUnavailable: 1,
+        },
+      },
+      template: {
+        metadata: {
+          labels: {
+            'app.kubernetes.io/name': 'metrics-server',
+            'app.kubernetes.io/component': 'metrics-server',
+          } + cfg.commonLabels,
+        },
+        spec: {
+          affinity: {
+            podAntiAffinity: {
+              requiredDuringSchedulingIgnoredDuringExecution: [
+                {
+                  labelSelector: {
+                    matchLabels: {
+                      'app.kubernetes.io/name': 'metrics-server',
+                      'app.kubernetes.io/component': 'metrics-server',
+                    } + cfg.commonLabels,
+                  },
+                  namespaces: [cfg.namespace],
+                  topologyKey: 'kubernetes.io/hostname',
+                },
+              ],
+            },
+          },
+          containers: [
+            {
+              image: cfg.image,
+              args: [
+                '--secure-port=10250',
+                '--kubelet-preferred-address-types=InternalIP,ExternalIP,Hostname',
+                '--kubelet-use-node-status-port',
+                '--metric-resolution=15s',
+                '--kubelet-certificate-authority=/etc/tls/kubelet-serving-ca-bundle/ca-bundle.crt',
+                '--kubelet-client-certificate=/etc/tls/metrics-client-certs/tls.crt',
+                '--kubelet-client-key=/etc/tls/metrics-client-certs/tls.key',
+                '--tls-cert-file=/etc/tls/private/tls.crt',
+                '--tls-private-key-file=/etc/tls/private/tls.key',
+                '--tls-cipher-suites=' + cfg.tlsCipherSuites,
+              ],
+              imagePullPolicy: 'IfNotPresent',
+              livenessProbe: {
+                failureThreshold: 3,
+                httpGet: {
+                  path: '/livez',
+                  port: 'https',
+                  scheme: 'HTTPS',
+                },
+                periodSeconds: 10,
+              },
+              name: 'metrics-server',
+              ports: [
+                {
+                  containerPort: 10250,
+                  name: 'https',
+                  protocol: 'TCP',
+                },
+              ],
+              readinessProbe: {
```

Comment:
It would be good to explain the probe settings (e.g. how it correlates to scrape intervals).